File: api/Models/Player.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace HockeyApi.Models
{
    public partial class Player
    {
        public Player()
        {
            Teams = new HashSet<Team>();
        }
        public int Id { get; set; }
        public int? Number { get; set; }
        public string Name { get; set; }
        public string Lastname { get; set; }
        public string Position { get; set; }
        public bool? IsCapitain { get; set; }

        [IgnoreDataMember]
        public ICollection<Team> Teams { get; set; }
    }
}

File: hockey-webapp/src/models/Player.ts
import { PositionType } from "./Common";

export interface Player {
  id: number;
  number?: number;
  name: string;
  lastname: string;
  position: PositionType;
  isCapitain: boolean;
}

export type PlayerPayload = Omit<Player, "id" | "teams">;
